rag/advanced_rag/harness/agent.py

```
@@ -0,0 +1,313 @@
+        phase=phase,
+        phase_hint=phase_config.get("tool_hint", ""),
+        max_cycles=mode.max_agent_cycles,
+    )
+    history = [{"role": "user", "content": f"Research task: {claim.description}\nBegin."}]
+
+    final_text = ""
+    try:
+        final_text = await agent_mdl.async_chat(system, history, {"temperature": 0.3})
+        if isinstance(final_text, tuple):
+            final_text = final_text[0]
+    except Exception:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)